vms/evm/warp/message/codec_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package message
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+// TestCodecSerialization tests the registration order changes in codec.go,
+// does not change, preventing unintended serialization format changes.
+func TestCodecSerialization(t *testing.T) {
```

Comment:
this is modeled (copy and pasted p much) from 

https://github.com/ava-labs/avalanchego/blob/d6386c131e828e246588a346936257ca67dc066e/vms/platformvm/block/serialization_test.go#L19-L121

As an aside, there are a lot of codec's in AvalancheGo - perhaps we could make tests against the codec manager interface and then do tests for all the codec to prevent changes

(like the same strategy Josh showed me for the database interface tests)